### PR TITLE
[DNM] luminous: core: _aio_log_start inflight overlap of 0x10000~1000 with [65536~4096]

### PR DESCRIPTION
**Pull request by jonsger**
http://tracker.ceph.com/issues/36755

There is quite some difference here to the commit in master. I adjusted it accordingly to the functions and syntax in luminous. The new test is green :)

### CODE REVIEW

**Comment by smithfarm** [comment]
@jonsger Your conflict resolution sections are an example for us all! Thanks.

File: src/test/objectstore/store_test.cc [hunk 1]
```diff
@@ -6855,6 +6855,90 @@ TEST_P(StoreTestSpecificAUSize, fsckOnUnalignedDevice2) {
   g_conf->apply_changes(NULL);
 }
 
+TEST_P(StoreTest, mergeRegionTest) {
```
Comment (ifed01) [inline src/test/objectstore/store_test.cc]: This test case (as well as above fsckOnUnaligned* which is currently missing) should be wrapped with #if defined(HAVE_LIBAIO).
That was the predecessor for (WITH_BLUESTORE) macros.

Comment (jonsger) [reply src/test/objectstore/store_test.cc]: Thanks for the hint. I did it, adjusted the conflicts description of the commit and force pushed the new commit.

**Comment by yuriw** [comment]
[wip-yuri3-testing-2018-11-27-2241-luminous](https://trello.com/c/ntdQZjkB/481-wip-yuri3-testing-2018-11-27-2241-luminous)

**Comment by jdurgin** [comment]
Looks like this is failing ceph-test-objectstore - from http://qa-proxy.ceph.com/teuthology/yuriw-2018-11-28_00:35:26-rados-wip-yuri3-testing-2018-11-27-2241-luminous-distro-basic-smithi/3286606/teuthology.log:

```
2018-11-28T07:48:17.849 INFO:teuthology.orchestra.run.smithi135.stderr:*** Caught signal (Aborted) **
2018-11-28T07:48:17.849 INFO:teuthology.orchestra.run.smithi135.stderr: in thread 7fa4f4f0f700 thread_name:finisher
2018-11-28T07:48:17.852 INFO:teuthology.orchestra.run.smithi135.stderr: ceph version 12.2.10-26-g740b499 (740b49941d5878dfed86a9b0c355fe42136b164e) luminous (stable)
2018-11-28T07:48:17.852 INFO:teuthology.orchestra.run.smithi135.stderr: 1: (()+0x540084) [0x55ffc66e5084]
2018-11-28T07:48:17.852 INFO:teuthology.orchestra.run.smithi135.stderr: 2: (()+0x11390) [0x7fa50e622390]
2018-11-28T07:48:17.852 INFO:teuthology.orchestra.run.smithi135.stderr: 3: (gsignal()+0x38) [0x7fa503c87428]
2018-11-28T07:48:17.852 INFO:teuthology.orchestra.run.smithi135.stderr: 4: (abort()+0x16a) [0x7fa503c8902a]
2018-11-28T07:48:17.852 INFO:teuthology.orchestra.run.smithi135.stderr: 5: (KernelDevice::_aio_log_start(IOContext*, unsigned long, unsigned long)+0x81c) [0x55ffc669f61c]
2018-11-28T07:48:17.853 INFO:teuthology.orchestra.run.smithi135.stderr: 6: (KernelDevice::aio_read(unsigned long, unsigned long, ceph::buffer::list*, IOContext*)+0x2c5) [0x55ffc66a0565]
2018-11-28T07:48:17.853 INFO:teuthology.orchestra.run.smithi135.stderr: 7: (BlueStore::_do_read(BlueStore::Collection*, boost::intrusive_ptr<BlueStore::Onode>, unsigned long, unsigned long, ceph::buffer::list&, unsigned int)+0x1c6b) [0x55ffc65ade5b]
2018-11-28T07:48:17.853 INFO:teuthology.orchestra.run.smithi135.stderr: 8: (BlueStore::read(boost::intrusive_ptr<ObjectStore::CollectionImpl>&, ghobject_t const&, unsigned long, unsigned long, ceph::buffer::list&, unsigned int)+0x4d7) [0x55ffc65b0417]
2018-11-28T07:48:17.853 INFO:teuthology.orchestra.run.smithi135.stderr: 9: (BlueStore::read(coll_t const&, ghobject_t const&, unsigned long, unsigned long, ceph::buffer::list&, unsigned int)+0x6f) [0x55ffc65756ff]
2018-11-28T07:48:17.853 INFO:teuthology.orchestra.run.smithi135.stderr: 10: (SyntheticWorkloadState::C_SyntheticOnStash::finish(int)+0x184) [0x55ffc6452794]
2018-11-28T07:48:17.853 INFO:teuthology.orchestra.run.smithi135.stderr: 11: (Context::complete(int)+0x9) [0x55ffc642b8c9]
2018-11-28T07:48:17.853 INFO:teuthology.orchestra.run.smithi135.stderr: 12: (Finisher::finisher_thread_entry()+0x460) [0x7fa50509fe80]
2018-11-28T07:48:17.853 INFO:teuthology.orchestra.run.smithi135.stderr: 13: (()+0x76ba) [0x7fa50e6186ba]
2018-11-28T07:48:17.854 INFO:teuthology.orchestra.run.smithi135.stderr: 14: (clone()+0x6d) [0x7fa503d5941d]```

**Comment by smithfarm** [comment]
@jonsger Ping

**Comment by smithfarm** [comment]
Closing as too high-risk for luminous at this stage of its life cycle.

**Pull request closed without merging**